specs/data_structures.md

```diff
@@ -320,31 +286,131 @@ A proof into an SMT is structured as:
 
 # Erasure Coding
 
+In order to enable trust-minimized light clients (i.e. light clients that do not rely on an honest majority of validating state assumption), it is critical that light clients can determine whether the data in each block is _available_ or not, without downloading the whole block itself. The technique used here was formally described in the paper [Fraud and Data Availability Proofs: Maximising Light Client Security and Scaling Blockchains with Dishonest Majorities](https://arxiv.org/abs/1809.09044).
```

Comment:
Technically in LazyLedger everyone uses data availability proofs, as there is no concept of a "light client" for the main chain.

Reply:
Hmm I guess in this case a "full node" would be one that fully validates the staking and fee part (basically, whatever's in the "reserved' namespace). We should probably write down specifically what different classes of nodes are and specifically what they are supposed to download, and what their assumptions and attack vectors are.